app/lib/actions/katello/repository/index_content.rb

```diff
@@ -31,6 +31,10 @@ def run
             output[:new_content][content_type.label] = new_count - initial_counts[content_type.label]
           end
         end
+
+        def finalize
+          ForemanTasks.async_task(OrphanCleanup::RemoveOrphanedContentUnits, {repo_id: input[:id]})
```

Comment:
I'm questioning now if we even need to destroy orphaned units after syncing.

For the orphaned filter rule problem, the solution is this: https://github.com/Katello/katello/blob/05f9de2ae38093f5e30e96eed21882671bab6ea2/app/services/katello/content_unit_indexer.rb#L155

Otherwise, I don't think orphaned content units should cause any other problems.

Plus, if a user syncs 20 repositories, they're going to get 20 orphan cleanup tasks running, which could eat up resources on their system.

Perhaps we should just rely on the weekly orphan cleanup?

If we do that, I just suggest that we change the following error messages to tell the users to run orphan cleanup instead of run a complete sync:

- https://github.com/Katello/katello/blob/master/app/services/katello/pulp3/repository.rb#L500
- https://github.com/Katello/katello/blob/master/app/services/katello/pulp3/repository.rb#L512
- https://github.com/Katello/katello/blob/master/app/services/katello/pulp3/repository/docker.rb#L73

Reply:
Actually, maybe we should change those error messages with your PR here regardless. Complete sync no longer deletes the orphaned units itself.